diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -43,76 +51,127 @@ class BlockStructureBraces : Rule("block-structure") {
+        catchBlocks.ifNotEmpty {
+            catchBlocks.forEach {
+                checkOpenBrace(it, BLOCK, configuration)
+                checkCloseBrace(it.findChildByType(BLOCK)!!, configuration)
+            }
+        }
+        if (finallyBlock != null) {
+            checkOpenBrace(finallyBlock, BLOCK, configuration)
+            checkCloseBrace(finallyBlock.findChildByType(BLOCK)!!, configuration)
         }
     }
 
-    private fun checkFor(node: ASTNode) {
-        checkOpenBrace(node, BODY)
-        checkCloseBrace(node.findChildByType(BODY)!!.findChildByType(BLOCK)!!)
+    private fun checkLoop(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
+        if (node.hasChildOfType(BODY) && node.findChildByType(BODY)!!.hasChildOfType(BLOCK)) {
+            checkOpenBrace(node, BODY, configuration)
+            checkCloseBrace(node.findChildByType(BODY)?.findChildByType(BLOCK), configuration)
+        }
     }
 
-    private fun checkWhen(node: ASTNode) {
-        checkOpenBrace(node, LBRACE)
-        checkCloseBrace(node)
+    private fun checkWhen(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
+        checkOpenBrace(node, LBRACE, configuration)
+        checkCloseBrace(node, configuration)
     }
 
-    private fun checkFun(node: ASTNode) {
+    private fun checkFun(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
         if (node.hasChildOfType(BLOCK)) {
-            checkOpenBrace(node, BLOCK)
-            checkCloseBrace(node.findChildByType(BLOCK)!!)
+            checkOpenBrace(node, BLOCK, configuration)
+            checkCloseBrace(node.findChildByType(BLOCK)!!, configuration)
         }
     }
 
-    private fun checkIf(node: ASTNode) {
+    private fun checkIf(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
         val ifPsi = node.psi as KtIfExpression
         val thenNode = ifPsi.then?.node
         val hasElseBranch = ifPsi.elseKeyword != null
         val elseNode = ifPsi.`else`?.node
-
-        checkOpenBrace(node, THEN)
-        checkCloseBrace(thenNode!!)
-        if (hasElseBranch && elseNode!!.elementType != IF) {
-            checkOpenBrace(node, ELSE)
-            checkCloseBrace(elseNode)
+        if (thenNode != null && thenNode.hasChildOfType(LBRACE)) {
+            checkOpenBrace(node, THEN, configuration)
+            checkCloseBrace(thenNode, configuration)
+        }
+        if (hasElseBranch && elseNode!!.elementType != IF && elseNode.hasChildOfType(LBRACE)) {
+            checkOpenBrace(node, ELSE, configuration)
+            checkCloseBrace(elseNode, configuration)
         }
     }
 
-    private fun checkOpenBrace(node: ASTNode, beforeType: IElementType) {
+    private fun checkOpenBrace(node: ASTNode, beforeType: IElementType, configuration: BlockStructureBracesConfiguration) {
+        if (!configuration.openBrace) return
         val braceSpace = node.findChildBefore(beforeType, WHITE_SPACE)
```

Comment:
Do you really need `findChildBefore` here? I think you might pass `BLOCK` node in `checkOpenBrace` fnction and then check it's `prevSibling`. Do you see any examples where it can not work?